src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,119 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        checkSize();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkSize();
+        if (index != size) {
+            indexCheck(index);
+        }
```

Comment:
```suggestion
        if (index == size) {
              add(value);
              return;
        }
        indexCheck(index);
```
You call checkSize() inside add